### PR TITLE
Add DatabricksShimVersion to unshimmed class list [databricks]

### LINKED ISSUE #4918: [BUG] databricks nightly build failed
**Describe the bug**
introduced in https://github.com/NVIDIA/spark-rapids/pull/4857

spark301db
```
scala.MatchError: 3.0.1-databricks (of class com.nvidia.spark.rapids.DatabricksShimVersion)
```

spark312db
```
java.lang.LinkageError: loader constraint violation: when resolving method "com.nvidia.spark.rapids.shims.spark312db.SparkShimServiceProvider$.VERSION()Lcom/nvidia/spark/rapids/DatabricksShimVersion;" the class loader (instance of org/apache/spark/util/MutableURLClassLoader) of the current class, com/nvidia/spark/rapids/shims/spark312db/SparkShimServiceProvider, and the class loader (instance of sun/misc/Launcher$AppClassLoader) for the method's defining class, com/nvidia/spark/rapids/shims/spark312db/SparkShimServiceProvider$, have different Class objects for the type com/nvidia/spark/rapids/DatabricksShimVersion used in the signature
```

### PR DESCRIPTION
**Pull request by razajafri**
~Reverting changes to the DB SparkShims as part of the #4857. This change was originally envisioned to avoid referencing the ServiceProvider, but in case of DB we are not reusing the SparkShimImpl so we can directly reference it.~

Talked to @tgravescs about this and confirmed the correct fix is to "unshim" the `DatabricksShimVersion` that way the class loader will pick up the correct class. 

fixes #4918 

Signed-off-by: Raza Jafri <rjafri@nvidia.com>

### CODE REVIEW

**Comment by razajafri** [comment]
@pxLi PTAL

**Comment by razajafri** [comment]
build

**Comment by razajafri** [comment]
build

**Comment by tgravescs** [comment]
I'm confused by the description here. I don't think this is the right fix. it appears the issue is with classloader and having 2 versions of the DatabricksSparkVersion, I think it needs to be added to the classes we don't shim.   We have com/nvidia/spark/rapids/SparkShimVersion and ShimVersion in there already.

**Comment by razajafri** [comment]
> 

You are right, I wanted to put up this patch to unblock the CI last night. I will put up another PR

**Comment by razajafri** [comment]
build

File: dist/unshimmed-common-from-spark301.txt [hunk 1]
```diff
@@ -19,6 +19,7 @@ com/nvidia/spark/rapids/ShimLoader*
 com/nvidia/spark/rapids/ShimVersion*
 com/nvidia/spark/rapids/SparkShimServiceProvider*
 com/nvidia/spark/rapids/SparkShimVersion*
+com/nvidia/spark/rapids/DatabricksShimVersion*
```
Comment (jlowe) [inline dist/unshimmed-common-from-spark301.txt:22]: Nit: This file is lexicographically ordered, would be good to keep that precedent.

**Comment by razajafri** [comment]
build

**Review by jlowe — APPROVED** [review]

**Review by tgravescs — APPROVED** [review]

**Pull request merged**